src/factorials.jl

```diff
@@ -29,6 +29,17 @@ function Base.factorial(n::T, k::T) where T<:Integer
     end
     return f
 end
+function Base.factorial(n::BigInt, k::BigInt)
+    if k < 0 || n < 0 || k > n
+        throw(DomainError((n, k), "n and k must be nonnegative with k ≤ n"))
+    end
+    f = BigInt(1)
```

Comment:
```suggestion
    f = BigInt(1)
    n = deepcopy(n)  # avoid mutating input
```

Currently, the `sub_ui!` function modifies `n`.
After calling the `factorial` function, `n == k`.

Reply:
Yeah you are right. Personally I would prefer a version which mutates `n` and leaves to the caller the responsibility of copying (for more efficiency), but then it's true that this version should be called `factorial!` then.